lib/SILGen/SILGenExpr.cpp

```diff
@@ -4436,7 +4438,6 @@ RValue RValueEmitter::visitIfExpr(IfExpr *E, SGFContext C) {
     
     // FIXME: We could avoid imploding and reexploding tuples here.
     Condition cond = SGF.emitCondition(E->getCondExpr(),
-                                       /*hasFalse*/ true,
```

Comment:
I think this commit needs to go before the previous one, no? Beyond that looks great!